File: snippets/screens/compose/signup_screen/3/signup_screen.kt
package com.example.auth.screens

import androidx.compose.foundation.layout.*
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.text.KeyboardActions
import androidx.compose.foundation.text.KeyboardOptions
import androidx.compose.foundation.verticalScroll
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Email
import androidx.compose.material.icons.filled.Lock
import androidx.compose.material.icons.filled.Person
import androidx.compose.material.icons.filled.Visibility
import androidx.compose.material.icons.filled.VisibilityOff
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.focus.FocusDirection
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.platform.LocalFocusManager
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.input.ImeAction
import androidx.compose.ui.text.input.KeyboardType
import androidx.compose.ui.text.input.PasswordVisualTransformation
import androidx.compose.ui.text.input.VisualTransformation
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.tooling.preview.Preview
import androidx.compose.ui.unit.dp
import androidx.lifecycle.viewmodel.compose.viewModel
import kotlinx.coroutines.launch

@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun SignupScreen(
    viewModel: SignupViewModel = viewModel(),
    onNavigateToLogin: () -> Unit = {},
    onSignupSuccess: () -> Unit = {}
) {
    val scrollState = rememberScrollState()
    val focusManager = LocalFocusManager.current
    val snackbarHostState = remember { SnackbarHostState() }
    val coroutineScope = rememberCoroutineScope()

    LaunchedEffect(viewModel.signupSuccess) {
        if (viewModel.signupSuccess) {
            coroutineScope.launch {
                snackbarHostState.showSnackbar("Account created successfully!")
            }
            onSignupSuccess()
        }
    }

    Scaffold(
        snackbarHost = { SnackbarHost(snackbarHostState) }
    ) { paddingValues ->
        Column(
            modifier = Modifier
                .fillMaxSize()
                .padding(paddingValues)
                .padding(24.dp)
                .verticalScroll(scrollState),
            horizontalAlignment = Alignment.CenterHorizontally
        ) {
            Spacer(modifier = Modifier.height(40.dp))

            // Header
            Text(
                text = "Create Account",
                style = MaterialTheme.typography.headlineMedium,
                fontWeight = FontWeight.Bold,
                textAlign = TextAlign.Center
            )

            Spacer(modifier = Modifier.height(8.dp))

            Text(
                text = "Sign up to get started",
                style = MaterialTheme.typography.bodyLarge,
                color = MaterialTheme.colorScheme.onSurfaceVariant,
                textAlign = TextAlign.Center
            )

            Spacer(modifier = Modifier.height(40.dp))

            // Name fields row
            Row(
                modifier = Modifier.fillMaxWidth(),
                horizontalArrangement = Arrangement.spacedBy(12.dp)
            ) {
                OutlinedTextField(
                    value = viewModel.firstName,
                    onValueChange = viewModel::updateFirstName,
                    label = { Text("First Name") },
                    leadingIcon = { Icon(Icons.Default.Person, contentDescription = null) },
                    isError = viewModel.firstNameError != null,
                    supportingText = viewModel.firstNameError?.let { { Text(it) } },
                    keyboardOptions = KeyboardOptions(
                        imeAction = ImeAction.Next
                    ),
                    keyboardActions = KeyboardActions(
                        onNext = { focusManager.moveFocus(FocusDirection.Next) }
                    ),
                    modifier = Modifier.weight(1f)
                )

                OutlinedTextField(
                    value = viewModel.lastName,
                    onValueChange = viewModel::updateLastName,
                    label = { Text("Last Name") },
                    leadingIcon = { Icon(Icons.Default.Person, contentDescription = null) },
                    isError = viewModel.lastNameError != null,
                    supportingText = viewModel.lastNameError?.let { { Text(it) } },
                    keyboardOptions = KeyboardOptions(
                        imeAction = ImeAction.Next
                    ),
                    keyboardActions = KeyboardActions(
                        onNext = { focusManager.moveFocus(FocusDirection.Next) }
                    ),
                    modifier = Modifier.weight(1f)
                )
            }

            Spacer(modifier = Modifier.height(16.dp))

            // Email field
            OutlinedTextField(
                value = viewModel.email,
                onValueChange = viewModel::updateEmail,
                label = { Text("Email") },
                leadingIcon = { Icon(Icons.Default.Email, contentDescription = null) },
                isError = viewModel.emailError != null,
                supportingText = viewModel.emailError?.let { { Text(it) } },
                keyboardOptions = KeyboardOptions(
                    keyboardType = KeyboardType.Email,
                    imeAction = ImeAction.Next
                ),
                keyboardActions = KeyboardActions(
                    onNext = { focusManager.moveFocus(FocusDirection.Next) }
                ),
                modifier = Modifier.fillMaxWidth()
            )

            Spacer(modifier = Modifier.height(16.dp))

            // Password field
            OutlinedTextField(
                value = viewModel.password,
                onValueChange = viewModel::updatePassword,
                label = { Text("Password") },
                leadingIcon = { Icon(Icons.Default.Lock, contentDescription = null) },
                trailingIcon = {
                    IconButton(onClick = { viewModel.togglePasswordVisibility() }) {
                        Icon(
                            imageVector = if (viewModel.passwordVisible) Icons.Default.Visibility
                                         else Icons.Default.VisibilityOff,
                            contentDescription = if (viewModel.passwordVisible) "Hide password" else "Show password"
                        )
                    }
                },
                visualTransformation = if (viewModel.passwordVisible) VisualTransformation.None
                                     else PasswordVisualTransformation(),
                isError = viewModel.passwordError != null,
                supportingText = {
                    Column {
                        viewModel.passwordError?.let { Text(it) }
                        if (viewModel.password.isNotBlank() && viewModel.passwordError == null) {
                            PasswordStrengthIndicator(viewModel.password)
                        }
                    }
                },
                keyboardOptions = KeyboardOptions(
                    keyboardType = KeyboardType.Password,
                    imeAction = ImeAction.Next
                ),
                keyboardActions = KeyboardActions(
                    onNext = { focusManager.moveFocus(FocusDirection.Next) }
                ),
                modifier = Modifier.fillMaxWidth()
            )

            Spacer(modifier = Modifier.height(16.dp))

            // Confirm password field
            OutlinedTextField(
                value = viewModel.confirmPassword,
                onValueChange = viewModel::updateConfirmPassword,
                label = { Text("Confirm Password") },
                leadingIcon = { Icon(Icons.Default.Lock, contentDescription = null) },
                trailingIcon = {
                    IconButton(onClick = { viewModel.toggleConfirmPasswordVisibility() }) {
                        Icon(
                            imageVector = if (viewModel.confirmPasswordVisible) Icons.Default.Visibility
                                         else Icons.Default.VisibilityOff,
                            contentDescription = if (viewModel.confirmPasswordVisible) "Hide password" else "Show password"
                        )
                    }
                },
                visualTransformation = if (viewModel.confirmPasswordVisible) VisualTransformation.None
                                     else PasswordVisualTransformation(),
                isError = viewModel.confirmPasswordError != null,
                supportingText = viewModel.confirmPasswordError?.let { { Text(it) } },
                keyboardOptions = KeyboardOptions(
                    keyboardType = KeyboardType.Password,
                    imeAction = ImeAction.Done
                ),
                keyboardActions = KeyboardActions(
                    onDone = {
                        focusManager.clearFocus()
                        if (viewModel.acceptTerms) viewModel.signup()
                    }
                ),
                modifier = Modifier.fillMaxWidth()
            )

            Spacer(modifier = Modifier.height(16.dp))

            // Terms and conditions checkbox
            Row(
                modifier = Modifier.fillMaxWidth(),
                verticalAlignment = Alignment.CenterVertically
            ) {
                Checkbox(
                    checked = viewModel.acceptTerms,
                    onCheckedChange = viewModel::updateAcceptTerms
                )

                Spacer(modifier = Modifier.width(8.dp))

                Text(
                    text = "I accept the Terms and Conditions and Privacy Policy",
                    style = MaterialTheme.typography.bodySmall,
                    modifier = Modifier.weight(1f)
                )
            }

            viewModel.error?.let { error ->
                Spacer(modifier = Modifier.height(16.dp))
                ErrorCard(message = error)
            }

            Spacer(modifier = Modifier.height(24.dp))

            // Sign up button
            Button(
                onClick = { viewModel.signup() },
                enabled = !viewModel.isLoading && viewModel.isFormValid && viewModel.acceptTerms,
                modifier = Modifier
                    .fillMaxWidth()
                    .height(56.dp)
            ) {
                if (viewModel.isLoading) {
                    CircularProgressIndicator(
                        modifier = Modifier.size(20.dp),
                        color = MaterialTheme.colorScheme.onPrimary
                    )
                } else {
                    Text("Sign Up")
                }
            }

            Spacer(modifier = Modifier.height(24.dp))

            // Sign in link
            Row(
                verticalAlignment = Alignment.CenterVertically
            ) {
                Text(
                    text = "Already have an account? ",
                    style = MaterialTheme.typography.bodyMedium
                )
                TextButton(onClick = onNavigateToLogin) {
                    Text("Sign In")
                }
            }

            Spacer(modifier = Modifier.height(40.dp))
        }
    }
}

@Composable
fun PasswordStrengthIndicator(password: String) {
    val strength = calculatePasswordStrength(password)
    val (color, text) = when (strength) {
        1 -> Pair(Color.Red, "Weak")
        2, 3 -> Pair(Color(0xFFFF9800), "Medium")
        4, 5 -> Pair(Color.Green, "Strong")
        else -> Pair(Color.Gray, "")
    }

    Column {
        LinearProgressIndicator(
            progress = strength / 5f,
            modifier = Modifier.fillMaxWidth(),
            color = color
        )
        Spacer(modifier = Modifier.height(4.dp))
        Text(
            text = "Password strength: $text",
            style = MaterialTheme.typography.bodySmall,
            color = color
        )
    }
}

private fun calculatePasswordStrength(password: String): Int {
    var score = 0
    if (password.length >= 8) score++
    if (password.any { it.isUpperCase() }) score++
    if (password.any { it.isLowerCase() }) score++
    if (password.any { it.isDigit() }) score++
    if (password.any { !it.isLetterOrDigit() }) score++
    return score
}

@Composable
fun ErrorCard(message: String) {
    Card(
        modifier = Modifier.fillMaxWidth(),
        colors = CardDefaults.cardColors(
            containerColor = MaterialTheme.colorScheme.errorContainer
        )
    ) {
        Row(
            modifier = Modifier.padding(16.dp),
            verticalAlignment = Alignment.CenterVertically
        ) {
            Icon(
                imageVector = Icons.Default.Person, // You would use an error icon here
                contentDescription = null,
                tint = MaterialTheme.colorScheme.onErrorContainer
            )
            Spacer(modifier = Modifier.width(12.dp))
            Text(
                text = message,
                style = MaterialTheme.typography.bodyMedium,
                color = MaterialTheme.colorScheme.onErrorContainer,
                modifier = Modifier.weight(1f)
            )
        }
    }
}

// ViewModel would be implemented separately
class SignupViewModel : androidx.lifecycle.ViewModel() {
    var firstName by mutableStateOf("")
        private set
    var lastName by mutableStateOf("")
        private set
    var email by mutableStateOf("")
        private set
    var password by mutableStateOf("")
        private set
    var confirmPassword by mutableStateOf("")
        private set
    var acceptTerms by mutableStateOf(false)
        private set

    var passwordVisible by mutableStateOf(false)
        private set
    var confirmPasswordVisible by mutableStateOf(false)
        private set

    var isLoading by mutableStateOf(false)
        private set
    var signupSuccess by mutableStateOf(false)
        private set
    var error by mutableStateOf<String?>(null)
        private set

    // Validation errors
    var firstNameError by mutableStateOf<String?>(null)
        private set
    var lastNameError by mutableStateOf<String?>(null)
        private set
    var emailError by mutableStateOf<String?>(null)
        private set
    var passwordError by mutableStateOf<String?>(null)
        private set
    var confirmPasswordError by mutableStateOf<String?>(null)
        private set

    val isFormValid: Boolean
        get() = firstName.isNotBlank() &&
                lastName.isNotBlank() &&
                isValidEmail(email) &&
                isValidPassword(password) &&
                password == confirmPassword

    fun updateFirstName(value: String) {
        firstName = value
        firstNameError = if (value.isBlank()) "First name is required" else null
    }

    fun updateLastName(value: String) {
        lastName = value
        lastNameError = if (value.isBlank()) "Last name is required" else null
    }

    fun updateEmail(value: String) {
        email = value
        emailError = if (value.isNotBlank() && !isValidEmail(value)) "Enter a valid email" else null
    }

    fun updatePassword(value: String) {
        password = value
        passwordError = if (value.isNotBlank() && !isValidPassword(value)) {
            "Password must be at least 8 characters with uppercase, lowercase, and number"
        } else null

        // Revalidate confirm password if it's filled
        if (confirmPassword.isNotBlank()) {
            confirmPasswordError = if (confirmPassword != value) "Passwords do not match" else null
        }
    }

    fun updateConfirmPassword(value: String) {
        confirmPassword = value
        confirmPasswordError = if (value.isNotBlank() && value != password) "Passwords do not match" else null
    }

    fun updateAcceptTerms(value: Boolean) {
        acceptTerms = value
        if (!value) {
            error = "Please accept the terms and conditions"
        } else {
            error = null
        }
    }

    fun togglePasswordVisibility() {
        passwordVisible = !passwordVisible
    }

    fun toggleConfirmPasswordVisibility() {
        confirmPasswordVisible = !confirmPasswordVisible
    }

    fun signup() {
        if (!isFormValid || !acceptTerms) {
            error = "Please complete all fields and accept terms"
            return
        }

        isLoading = true
        error = null

        // Simulate API call
        androidx.lifecycle.viewModelScope.launch {
            try {
                kotlinx.coroutines.delay(2000)
                signupSuccess = true
            } catch (e: Exception) {
                error = "Failed to create account. Please try again."
            } finally {
                isLoading = false
            }
        }
    }

    private fun isValidEmail(email: String): Boolean {
        return android.util.Patterns.EMAIL_ADDRESS.matcher(email).matches()
    }

    private fun isValidPassword(password: String): Boolean {
        return password.length >= 8 &&
                password.any { it.isUpperCase() } &&
                password.any { it.isLowerCase() } &&
                password.any { it.isDigit() }
    }
}

@Preview(showBackground = true)
@Composable
fun SignupScreenPreview() {
    MaterialTheme {
        SignupScreen()
    }
}

File: snippets/screens/swiftui/signup_screen/3/signup_screen.swift
import SwiftUI
import Combine

struct SignupScreen: View {
    @StateObject private var viewModel = SignupViewModel()
    @State private var acceptTerms = false

    var body: some View {
        NavigationView {
            ScrollView {
                VStack(spacing: 20) {
                    Spacer()
                        .frame(height: 40)

                    // Header
                    VStack(spacing: 8) {
                        Text("Create Account")
                            .font(.largeTitle)
                            .fontWeight(.bold)

                        Text("Sign up to get started")
                            .font(.body)
                            .foregroundColor(.secondary)
                    }
                    .padding(.bottom, 40)

                    // Form
                    VStack(spacing: 16) {
                        // Name fields
                        HStack(spacing: 12) {
                            VStack(alignment: .leading, spacing: 8) {
                                TextField("First Name", text: $viewModel.firstName)
                                    .textFieldStyle(CustomTextFieldStyle())
                                    .textContentType(.givenName)
                                    .autocorrectionDisabled()

                                if let error = viewModel.firstNameError {
                                    Text(error)
                                        .font(.caption)
                                        .foregroundColor(.red)
                                }
                            }

                            VStack(alignment: .leading, spacing: 8) {
                                TextField("Last Name", text: $viewModel.lastName)
                                    .textFieldStyle(CustomTextFieldStyle())
                                    .textContentType(.familyName)
                                    .autocorrectionDisabled()

                                if let error = viewModel.lastNameError {
                                    Text(error)
                                        .font(.caption)
                                        .foregroundColor(.red)
                                }
                            }
                        }

                        // Email field
                        VStack(alignment: .leading, spacing: 8) {
                            TextField("Email", text: $viewModel.email)
                                .textFieldStyle(CustomTextFieldStyle(icon: "envelope"))
                                .keyboardType(.emailAddress)
                                .textContentType(.emailAddress)
                                .autocorrectionDisabled()
                                .autocapitalization(.none)

                            if let error = viewModel.emailError {
                                Text(error)
                                    .font(.caption)
                                    .foregroundColor(.red)
                            }
                        }

                        // Password field
                        VStack(alignment: .leading, spacing: 8) {
                            SecureField("Password", text: $viewModel.password)
                                .textFieldStyle(CustomTextFieldStyle(icon: "lock"))
                                .textContentType(.newPassword)

                            if let error = viewModel.passwordError {
                                Text(error)
                                    .font(.caption)
                                    .foregroundColor(.red)
                            } else if !viewModel.password.isEmpty {
                                PasswordStrengthView(password: viewModel.password)
                            }
                        }

                        // Confirm password field
                        VStack(alignment: .leading, spacing: 8) {
                            SecureField("Confirm Password", text: $viewModel.confirmPassword)
                                .textFieldStyle(CustomTextFieldStyle(icon: "lock"))
                                .textContentType(.newPassword)

                            if let error = viewModel.confirmPasswordError {
                                Text(error)
                                    .font(.caption)
                                    .foregroundColor(.red)
                            }
                        }
                    }

                    // Terms and conditions
                    VStack(spacing: 16) {
                        HStack(alignment: .top, spacing: 12) {
                            Button(action: {
                                acceptTerms.toggle()
                            }) {
                                Image(systemName: acceptTerms ? "checkmark.square.fill" : "square")
                                    .foregroundColor(acceptTerms ? .blue : .secondary)
                            }

                            Text("I accept the **Terms and Conditions** and **Privacy Policy**")
                                .font(.caption)
                                .multilineTextAlignment(.leading)
                        }

                        if let error = viewModel.error {
                            ErrorView(message: error)
                        }
                    }

                    // Sign up button
                    Button(action: {
                        if acceptTerms {
                            viewModel.signUp()
                        } else {
                            viewModel.error = "Please accept the terms and conditions"
                        }
                    }) {
                        if viewModel.isLoading {
                            ProgressView()
                                .progressViewStyle(CircularProgressViewStyle(tint: .white))
                        } else {
                            Text("Sign Up")
                        }
                    }
                    .buttonStyle(PrimaryButtonStyle())
                    .disabled(viewModel.isLoading || !viewModel.isFormValid || !acceptTerms)

                    // Sign in link
                    HStack {
                        Text("Already have an account?")
                            .foregroundColor(.secondary)

                        Button("Sign In") {
                            // Navigate to login
                        }
                    }
                    .padding(.top, 24)

                    Spacer()
                        .frame(height: 40)
                }
                .padding(.horizontal, 24)
            }
        }
        .navigationBarHidden(true)
    }
}

class SignupViewModel: ObservableObject {
    @Published var firstName = ""
    @Published var lastName = ""
    @Published var email = ""
    @Published var password = ""
    @Published var confirmPassword = ""
    @Published var isLoading = false
    @Published var error: String?

    // Validation errors
    @Published var firstNameError: String?
    @Published var lastNameError: String?
    @Published var emailError: String?
    @Published var passwordError: String?
    @Published var confirmPasswordError: String?

    private var cancellables = Set<AnyCancellable>()

    var isFormValid: Bool {
        return !firstName.isEmpty &&
               !lastName.isEmpty &&
               isValidEmail(email) &&
               isValidPassword(password) &&
               password == confirmPassword
    }

    init() {
        setupValidation()
    }

    private func setupValidation() {
        // Real-time validation
        $firstName
            .debounce(for: 0.3, scheduler: RunLoop.main)
            .sink { [weak self] value in
                self?.firstNameError = value.isEmpty ? "First name is required" : nil
            }
            .store(in: &cancellables)

        $lastName
            .debounce(for: 0.3, scheduler: RunLoop.main)
            .sink { [weak self] value in
                self?.lastNameError = value.isEmpty ? "Last name is required" : nil
            }
            .store(in: &cancellables)

        $email
            .debounce(for: 0.5, scheduler: RunLoop.main)
            .sink { [weak self] value in
                if !value.isEmpty && !self?.isValidEmail(value) ?? true {
                    self?.emailError = "Enter a valid email"
                } else {
                    self?.emailError = nil
                }
            }
            .store(in: &cancellables)

        $password
            .debounce(for: 0.3, scheduler: RunLoop.main)
            .sink { [weak self] value in
                if !value.isEmpty && !self?.isValidPassword(value) ?? true {
                    self?.passwordError = "Password must be at least 8 characters with uppercase, lowercase, and number"
                } else {
                    self?.passwordError = nil
                }
            }
            .store(in: &cancellables)

        $confirmPassword
            .debounce(for: 0.3, scheduler: RunLoop.main)
            .sink { [weak self] value in
                if !value.isEmpty && value != self?.password {
                    self?.confirmPasswordError = "Passwords do not match"
                } else {
                    self?.confirmPasswordError = nil
                }
            }
            .store(in: &cancellables)
    }

    private func isValidEmail(_ email: String) -> Bool {
        let emailRegex = #"^[\w\-\.]+@([\w\-]+\.)+[\w\-]{2,4}$"#
        return NSPredicate(format: "SELF MATCHES %@", emailRegex).evaluate(with: email)
    }

    private func isValidPassword(_ password: String) -> Bool {
        return password.count >= 8 &&
               password.range(of: "[A-Z]", options: .regularExpression) != nil &&
               password.range(of: "[a-z]", options: .regularExpression) != nil &&
               password.range(of: "[0-9]", options: .regularExpression) != nil
    }

    func signUp() {
        guard isFormValid else { return }

        isLoading = true
        error = nil

        // Simulate API call
        DispatchQueue.main.asyncAfter(deadline: .now() + 2) {
            self.isLoading = false
            // Handle success/failure
            // Navigate to home or show error
        }
    }
}

struct PasswordStrengthView: View {
    let password: String

    private var strength: PasswordStrength {
        var score = 0

        if password.count >= 8 { score += 1 }
        if password.range(of: "[A-Z]", options: .regularExpression) != nil { score += 1 }
        if password.range(of: "[a-z]", options: .regularExpression) != nil { score += 1 }
        if password.range(of: "[0-9]", options: .regularExpression) != nil { score += 1 }
        if password.range(of: "[^A-Za-z0-9]", options: .regularExpression) != nil { score += 1 }

        switch score {
        case 0...1: return .weak
        case 2...3: return .medium
        default: return .strong
        }
    }

    var body: some View {
        VStack(alignment: .leading, spacing: 4) {
            ProgressView(value: Double(strength.rawValue), total: 3.0)
                .progressViewStyle(LinearProgressViewStyle(tint: strength.color))
                .frame(height: 4)

            Text("Password strength: \(strength.text)")
                .font(.caption)
                .foregroundColor(strength.color)
        }
    }
}

enum PasswordStrength: Int {
    case weak = 1
    case medium = 2
    case strong = 3

    var text: String {
        switch self {
        case .weak: return "Weak"
        case .medium: return "Medium"
        case .strong: return "Strong"
        }
    }

    var color: Color {
        switch self {
        case .weak: return .red
        case .medium: return .orange
        case .strong: return .green
        }
    }
}

// Custom styles
struct CustomTextFieldStyle: TextFieldStyle {
    let icon: String?

    init(icon: String? = nil) {
        self.icon = icon
    }

    func _body(configuration: TextField<_Label>) -> some View {
        HStack {
            if let icon = icon {
                Image(systemName: icon)
                    .foregroundColor(.secondary)
                    .frame(width: 20)
            }

            configuration
        }
        .padding(.horizontal, 16)
        .padding(.vertical, 12)
        .background(Color(.systemGray6))
        .cornerRadius(12)
        .overlay(
            RoundedRectangle(cornerRadius: 12)
                .stroke(Color(.systemGray4), lineWidth: 1)
        )
    }
}

struct PrimaryButtonStyle: ButtonStyle {
    func makeBody(configuration: Configuration) -> some View {
        configuration.label
            .frame(maxWidth: .infinity)
            .frame(height: 56)
            .background(configuration.isPressed ? Color.blue.opacity(0.8) : Color.blue)
            .foregroundColor(.white)
            .cornerRadius(12)
            .scaleEffect(configuration.isPressed ? 0.98 : 1.0)
            .animation(.easeInOut(duration: 0.1), value: configuration.isPressed)
    }
}

struct ErrorView: View {
    let message: String

    var body: some View {
        HStack {
            Image(systemName: "exclamationmark.circle")
                .foregroundColor(.red)

            Text(message)
                .font(.caption)
                .foregroundColor(.red)
                .multilineTextAlignment(.leading)

            Spacer()
        }
        .padding(.horizontal, 12)
        .padding(.vertical, 8)
        .background(Color.red.opacity(0.1))
        .cornerRadius(8)
        .overlay(
            RoundedRectangle(cornerRadius: 8)
                .stroke(Color.red.opacity(0.3), lineWidth: 1)
        )
    }
}

#Preview {
    SignupScreen()
}